### DISCUSSION TITLE
[Rejected] Proposal: Renaming Entities to Actors

### DISCUSSION THREAD

**Discussion by xezno** [xezno/mocha#48] [Announcements]
## What

I'm thinking of renaming entities to actors. There are no functional changes to this, it's just a rename.
 - Anything prefixed or suffixed with `Entity` is replaced with `Actor`, e.g. `BaseEntity` -> `BaseActor`
 - On the native side: 
   - `g_entityDictionary` also becomes `g_actorManager`
   - Entity flags become `ACTOR_FLAG_X` instead of `ENTITY_X`

## Why

- Actor is pretty close to the beginning of the alphabet (closer than Entity)
- It's the terminology Unreal uses, which is rapidly becoming one of the biggest engines in the world... this would make it easier for people to switch to/from Mocha

## When

Ideally this would happen after the "networking" branch gets merged, because that branch will probably change a lot of entity stuff. That being said, there's already an "actors" branch if you want to go check it out.

---

If you like this change give this post a thumbs up or something, otherwise give it a thumbs down, and we'll go from there. If you have anything else to say please feel free below.

**Comment by Xenthio** [comment]
I feel like Actor implies that something is on screen, So in my opinion the name Actor would fit better for a ModelEntity rename or something, instead of all entities

**Reply by xezno** [reply]
I agree

**Comment by lannior** [comment]
"Entity" is vague and that's why it's better in my opinion.
"Actor" implies something living, on screen.

I don't want my gun, ammo box or treasure chest to be an actor. ModelActor also sounds considerably worse than ModelEntity, since actor already sorta implies having a model?

Not a big fan of renaming ModelEntity to Actor either, it will just make things more confusing.

**Reply by xezno** [reply]
Noted
